### ISSUE TITLE
Understanding the complete project workflow

### ISSUE THREAD

**Issue by sks444** [HyphaApp/hypha#2580]
Hello Team,

@frjo @fourthletter @ei8fdb 

Creating this issue to understand the complete project workflow and requirements at one place. This will help me in planing/prioritising work and in collaborating with a new developer that I will work with. Later on we can create a WiKi from these information: which would be helpful for anyone who is new to the projects part of Hypha system.

I'll be describing what we have and what we're planning, according to my understanding.

Please let me know if something is missing or anything that needs to be deprecated.

---

 A project gets created when a [submission is approved](https://github.com/HyphaApp/hypha/blob/d02f7a309446bf15e66b5aed6757ddc76e6ed42f/hypha/apply/determinations/views.py#L402). There is also an issue to create projects without an application https://github.com/HyphaApp/hypha/issues/2447

On automatic creation of project we also initialise few of the [vendor information](https://github.com/HyphaApp/hypha/blob/d02f7a309446bf15e66b5aed6757ddc76e6ed42f/hypha/apply/projects/models/project.py#L212).

Once a project is created we see the following informations on projects page:

<img width="1407" alt="Screenshot 2021-10-12 at 1 03 12 PM" src="https://user-images.githubusercontent.com/22263040/136912207-674f912d-b8dc-4f74-b40d-1c552f760b0c.png">

Where there is an option to upload supporting documents. This allows applicant to upload documents of different category:

<img width="417" alt="Screenshot 2021-10-12 at 1 09 04 PM" src="https://user-images.githubusercontent.com/22263040/136912905-66ede0d4-1319-49bc-81df-1e0278877fa2.png">

**Do we still need this feature or anything that needs to be modified?**

Once a project is created a lead must be assigned.

Then we see the option to setup `vendor/contractor`: where vendor fill up their banking information and upload tax documents etc.

And an option to add `Approval Form` which I think is related to the [contract creation and management process ](https://github.com/HyphaApp/hypha/issues/2476) This is something which needs to be completed before vendor can generate invoices and start receiving funding. We are targeting this work once the payment process is complete.

Once a contract is uploaded-signed and approved we see the option to add invoices.

Project invoicing process is almost complete at https://github.com/HyphaApp/hypha/pull/2450; That means we can remove the old implementation of Payment Requests, which is work in progress: https://github.com/HyphaApp/hypha/issues/2579. 

We would still need to work on the invoicing process more. E.g. How will the flow will be with different level of user PM/Finance-1/Finance-2 and for a project which doesn't have any deliverables.

If a project has deliverable, a way to add them on an invoice is already done: https://github.com/HyphaApp/hypha/pull/2556

Different issues related to invoices that needs to be addressed:

Finance level 1 user needs to approve a fixed-price invoice https://github.com/HyphaApp/hypha/issues/2444
PM can request modifications to fixed-price invoice https://github.com/HyphaApp/hypha/issues/2453
Finance level 1 can request modifications to a fixed-price invoice  https://github.com/HyphaApp/hypha/issues/2454
Finance level 2 can request modifications to a fixed-price invoice https://github.com/HyphaApp/hypha/issues/2457
Finance level 2 user can review and approve a fixed-price invoice https://github.com/HyphaApp/hypha/issues/2445

Other related tasks for projects:

Add a summary field to projects https://github.com/HyphaApp/hypha/issues/2448
Update information displayed in project Activity feed https://github.com/HyphaApp/hypha/issues/2458
Activity and Communication tabs should indicate unread messages and updates https://github.com/HyphaApp/hypha/issues/2479
[Meta] Project Page Notifications https://github.com/HyphaApp/hypha/issues/2477
Add files tab to individual project view for finance user https://github.com/HyphaApp/hypha/issues/2414
Update invoice table on dashboard and add a invoice/all page https://github.com/HyphaApp/hypha/issues/2399

### Integration of different third party payment services

Currently we are working on the SageIntAcct integration for open tech fund.

[META] IntAcct API extension  https://github.com/OpenTechFund/apply-app/issues/1
Hypha will only create invoices in IntAcct, what info is needed from IntAcct? https://github.com/HyphaApp/hypha/issues/2578

The integration needs to be done in such a way that it is easy to configure and add support to other payment services for other organization.

**Comment by Erioldoesdesign** [comment]
@sks444 Some of these workflows are discussed in both adopters monthly meetings (we covered ARDC in September) and also briefly discussed in Maintainer meetings. Next Adopter call will cover the adopter DDP's workflow broadly: https://we.hypha.app/t/monthly-adopters-meeting-2021-oct-27th/154 

OTF and Reset's workflow has been investigated by Bernard (design) and captured here on Miro boards: https://miro.com/app/board/o9J_lTav8ps=/ 
https://miro.com/app/board/o9J_l5Ldp0I=/
https://miro.com/app/board/o9J_koOOBcg=/

Are these useful?

**Comment by sks444** [comment]
Thanks @Erioldoesdesign, that will be useful for onboarding someone to Hypha.

However, this issue is specifically related to projects part of hypha. E.g. All the things that happens to a submission when it is approved...we create a project from that submission.

**Comment by Erioldoesdesign** [comment]
> Thanks @Erioldoesdesign, that will be useful for onboarding someone to Hypha.
> 
> However, this issue is specifically related to projects part of hypha. E.g. All the things that happens to a submission when it is approved...we create a project from that submission.

Okay good to know, this sounds like a hypha instance specific request for workflow walkthrough for projects which I also want to understand and know more about. I requested this walkthrough via email to OTF today and I'll see when it might happen. :)

**Comment by frjo** [comment]
@ei8fdb and I walked through this today to find and missing step etc. @ei8fdb is documenting everything and we are going to create all needed issues shortly.

**Comment by ei8fdb** [comment]
@sks444 Sorry it's taken so long to reply. I think the lack of clarity might be due to the lack of end-to-end workflow being documented. And I guess that might be down to me. Sorry for that.

I thought each of the User Stories and GH issues would have been enough. Not a problem! Working remotely isn't easy. I know that.

## To answer your questions from above

I'll quote your questions from above and answer them 1-by-1.

> Where there is an option to upload supporting documents. This allows applicant to upload documents of different category:

> <img width="417" alt="Screenshot 2021-10-12 at 1 09 04 PM" src="https://user-images.githubusercontent.com/22263040/136912905-66ede0d4-1319-49bc-81df-1e0278877fa2.png">

> **Do we still need this feature or anything that needs to be modified?**

Yes, this is still needed.

> contract creation and management process This is something which needs to be completed before vendor can generate invoices and start receiving funding.

Exacly correct.

> We are targeting this work once the payment process is complete.

Exacly correct.

> Once a contract is uploaded-signed and approved we see the option to add invoices.

Exacly correct.

> ...work on the invoicing process more. E.g. How will the flow will be with different level of user PM/Finance-1/Finance-2 

The flows for user:
- PM is documented in ticket 
- Finance-1 is documented in https://github.com/HyphaApp/hypha/issues/2444
- Finance-2 is documented in https://github.com/HyphaApp/hypha/issues/2445

> ...and for a project which doesn't have any deliverables.

I think I am right 99.999% that **all OTF projects** (with fixed price contract i.e. the type we are currently support with the finance work) will have at least 1 deliverable - so this shouldn't be a concern for the moment.   


## The whole end-to-end workflow
This morning @frjo and I mapped out the OTF workflow from submitting a concept note as far as the PM reviewing an invoice. 

- [ ] You will find the Concept Note Submission here - not yet done.
- [ ] You will find the Concept Note Review here - not yet done.
- [ ] You will find the Full Proposal Submission here - not yet done.
- [ ] You will find the Full Proposal Review here - not yet done.
- [x] You will find the Contractor Setup and Approval workflow here: - https://miro.com/app/board/o9J_lq3_-sY=/?moveToWidget=3074457365658844059&cot=14
- [x] You can find the Finance workflow here - https://miro.com/app/board/o9J_lq3_-sY=/?moveToWidget=3074457365658152075&cot=14

It contains the 1) contractor submits invoice, 2) PM reviews invoice work. 

@sks444 I've not added the Finance 1, Finance 2 user approves the invoice. Would you like me to do that? If so I will use the wireframes from each Github issue.

@sks444 Does that give you better understanding? If not, please let me know what I've missed and I'll add it.

**Comment by Erioldoesdesign** [comment]
@emlini I feel like some of this info is useful for documentation - perhaps not immediately, but I think it's worth you reading through this issue and the subsequent comments and projects workflow mapping from Bernard.

**Comment by fourthletter** [comment]
@Erioldoesdesign 

sks444 [mentioned above ](https://github.com/HyphaApp/hypha/issues/2580#issue-1023541940) they would create a "WiKi which would be helpful for anyone who is new to the projects part of Hypha system" -- Is a Wiki the best place for documentation OR could it go into https://docs.hypha.app/?

**Comment by frjo** [comment]
It should definitely go in the docs.

**Comment by fourthletter** [comment]
@sks444 I'm touching base to see if you have time over the next few weeks to write up security documentation for these features? For example, the type of testing you did related to security. Thank you!
